### ISSUE TITLE
Runtime: drop `[].slice.call( arguments, 0 )`?

### ISSUE THREAD

**Issue by rxaviers** [jquery/globalize#485]
In the normal modules, `arguments` is cloned using slice (e.g., https://github.com/jquery/globalize/blob/fix-398-runtime/src/message.js#L64), so it's not subject to further changes (e.g., https://github.com/jquery/globalize/blob/fix-398-runtime/src/message.js#L70). Therefore, on each respective runtime module the same procedure must happen again (e.g., https://github.com/jquery/globalize/blob/fix-398-runtime/src/message-runtime.js#L15) or the hash output would be different (given it's based on JSON.stringify and stringifying arguments has different output than stringifying an Array).

It should be possible to avoid all that slice thing (and instead use `arguments` directly in the hashing function) if no method (like the one exemplified above) changes any passed variable. Therefore, arguments would never change, therefore there would be no need to clone them.

The goal of this issue is to investigate if this proposed change would simplify the code and is worth making.

**Comment by jzaefferer** [comment]
Note that `slide` is hardly a clone. It only copies the references, so any things to objects themselves would still propagate. If that's a real concern, we should do proper deep cloning instead. Currently I don't see why that should be a concern.

**Comment by rxaviers** [comment]
Please try removing the slices and run the example

On Sunday, August 23, 2015, Jörn Zaefferer <notifications@github.com> wrote:

> Note that slide is hardly a clone. It only copies the references, so any
> things to objects themselves would still propagate. If that's a real
> concern, we should do proper deep cloning instead. Currently I don't see
> why that should be a concern.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jquery/globalize/issues/485#issuecomment-133817796>.
>


-- 
+55 (16) 98138-1583, +1 (415) 568-5854, skype: rxaviers
http://rafael.xavier.blog.br

**Comment by rxaviers** [comment]
Nope. Arguments aren't modified, but reassigned. So, deep cloning isn't needed.
